// @azure/arm-resources/dist/commonjs/models/index.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnownTagsPatchOperation = exports.KnownExportTemplateOutputFormat = exports.KnownExtendedLocationType = exports.KnownProviderAuthorizationConsentState = exports.KnownAliasPathAttributes = exports.KnownAliasPathTokenType = void 0;
/** Known values of {@link AliasPathTokenType} that the service accepts. */
var KnownAliasPathTokenType;
(function (KnownAliasPathTokenType) {
    /** The token type is not specified. */
    KnownAliasPathTokenType["NotSpecified"] = "NotSpecified";
    /** The token type can be anything. */
    KnownAliasPathTokenType["Any"] = "Any";
    /** The token type is string. */
    KnownAliasPathTokenType["String"] = "String";
    /** The token type is object. */
    KnownAliasPathTokenType["Object"] = "Object";
    /** The token type is array. */
    KnownAliasPathTokenType["Array"] = "Array";
    /** The token type is integer. */
    KnownAliasPathTokenType["Integer"] = "Integer";
    /** The token type is number. */
    KnownAliasPathTokenType["Number"] = "Number";
    /** The token type is boolean. */
    KnownAliasPathTokenType["Boolean"] = "Boolean";
})(KnownAliasPathTokenType || (exports.KnownAliasPathTokenType = KnownAliasPathTokenType = {}));
/** Known values of {@link AliasPathAttributes} that the service accepts. */
var KnownAliasPathAttributes;
(function (KnownAliasPathAttributes) {
    /** The token that the alias path is referring to has no attributes. */
    KnownAliasPathAttributes["None"] = "None";
    /** The token that the alias path is referring to is modifiable by policies with 'modify' effect. */
    KnownAliasPathAttributes["Modifiable"] = "Modifiable";
})(KnownAliasPathAttributes || (exports.KnownAliasPathAttributes = KnownAliasPathAttributes = {}));
/** Known values of {@link ProviderAuthorizationConsentState} that the service accepts. */
var KnownProviderAuthorizationConsentState;
(function (KnownProviderAuthorizationConsentState) {
    /** NotSpecified */
    KnownProviderAuthorizationConsentState["NotSpecified"] = "NotSpecified";
    /** Required */
    KnownProviderAuthorizationConsentState["Required"] = "Required";
    /** NotRequired */
    KnownProviderAuthorizationConsentState["NotRequired"] = "NotRequired";
    /** Consented */
    KnownProviderAuthorizationConsentState["Consented"] = "Consented";
})(KnownProviderAuthorizationConsentState || (exports.KnownProviderAuthorizationConsentState = KnownProviderAuthorizationConsentState = {}));
/** Known values of {@link ExtendedLocationType} that the service accepts. */
var KnownExtendedLocationType;
(function (KnownExtendedLocationType) {
    /** EdgeZone */
    KnownExtendedLocationType["EdgeZone"] = "EdgeZone";
})(KnownExtendedLocationType || (exports.KnownExtendedLocationType = KnownExtendedLocationType = {}));
/** Known values of {@link ExportTemplateOutputFormat} that the service accepts. */
var KnownExportTemplateOutputFormat;
(function (KnownExportTemplateOutputFormat) {
    /** Json */
    KnownExportTemplateOutputFormat["Json"] = "Json";
    /** Bicep */
    KnownExportTemplateOutputFormat["Bicep"] = "Bicep";
})(KnownExportTemplateOutputFormat || (exports.KnownExportTemplateOutputFormat = KnownExportTemplateOutputFormat = {}));
/** Known values of {@link TagsPatchOperation} that the service accepts. */
var KnownTagsPatchOperation;
(function (KnownTagsPatchOperation) {
    /** The 'replace' option replaces the entire set of existing tags with a new set. */
    KnownTagsPatchOperation["Replace"] = "Replace";
    /** The 'merge' option allows adding tags with new names and updating the values of tags with existing names. */
    KnownTagsPatchOperation["Merge"] = "Merge";
    /** The 'delete' option allows selectively deleting tags based on given names or name\/value pairs. */
    KnownTagsPatchOperation["Delete"] = "Delete";
})(KnownTagsPatchOperation || (exports.KnownTagsPatchOperation = KnownTagsPatchOperation = {}));
